cmd/exporter/main.go

```diff
@@ -227,7 +232,39 @@ func export() error {
 		},
 	}
 
-	if err := exporter.Export(layersDir, appDir, appImage, runImageID.String(), analyzedMD.Metadata, imageNames[1:], launcherConfig, stackMD); err != nil {
+	var cacheStore lifecycle.Cache
+	if cacheImageTag != "" {
+		origCacheImage, err := remote.NewImage(
+			cacheImageTag,
+			auth.DefaultEnvKeychain(),
+			remote.FromBaseImage(cacheImageTag),
+		)
+		if err != nil {
+			return cmd.FailErr(err, "accessing cache image")
+		}
+
+		emptyImage, err := remote.NewImage(
+			cacheImageTag,
+			auth.DefaultEnvKeychain(),
+			remote.WithPreviousImage(cacheImageTag),
+		)
+		if err != nil {
+			return cmd.FailErr(err, "creating new cache image")
+		}
+
+		cacheStore = cache.NewImageCache(
+			origCacheImage,
+			emptyImage,
+		)
+	} else {
+		var err error
+		cacheStore, err = cache.NewVolumeCache(cacheDir)
+		if err != nil {
+			return cmd.FailErr(err, "create volume cache")
+		}
+	}
+
+	if err := exporter.Export(layersDir, appDir, appImage, runImageID.String(), analyzedMD.Metadata, imageNames[1:], launcherConfig, stackMD, cacheStore); err != nil {
```

Comment:
A shocking number of args. If really, really simple, I suggest a struct.

Reply:
There are 41 calls to this in the exporter_test, so not that simple unfortunately. I think it should be a separate PR.